connectors/modeshape-connector-git/src/main/java/org/modeshape/connector/git/GitTree.java

```diff
@@ -85,26 +85,23 @@ public Document execute( Repository repository,
             try {
                 RevCommit commit = walker.parseCommit(objId);
 
-                // could happen if not enough permissions, for example
-                if (commit != null) {
-                    // Add the properties for this node ...
-                    String committer = commiterName(commit);
-                    String author = authorName(commit);
-                    DateTime committed = values.dateFrom(commit.getCommitTime());
-                    writer.setPrimaryType(GitLexicon.FOLDER);
-                    writer.addProperty(JcrLexicon.CREATED, committed);
-                    writer.addProperty(JcrLexicon.CREATED_BY, committer);
-                    writer.addProperty(GitLexicon.OBJECT_ID, objId.name());
-                    writer.addProperty(GitLexicon.AUTHOR, author);
-                    writer.addProperty(GitLexicon.COMMITTER, committer);
-                    writer.addProperty(GitLexicon.COMMITTED, committed);
-                    writer.addProperty(GitLexicon.TITLE, commit.getShortMessage());
-                    writer.addProperty(GitLexicon.HISTORY, GitHistory.referenceToHistory(objId, branchOrTagOrObjectId, values));
-                    writer.addProperty(GitLexicon.DETAIL, GitCommitDetails.referenceToCommit(objId, values));
-                }
+                // Add the properties for this node ...
```

Comment:
from the JavaDoc of `parseCommit`
>Returns:
reference to the commit object. Never null.

Not sure why the `null` checks were ever there...